### PR TITLE
Log slow queries if they pass a configurable threshold

### LINKED ISSUE #6429: Log slow queries
### Feature Request

The database should be able to log queries that exceed a threshold to assist operators with identifying potential problematic queries.  This setting should log the slow queries even if query logging is disabled.

We currently have a `query-timeout` setting that will kill a long running query.  We also have `show queries` that will report the duration of the query.  These are similar ideas, but solve slightly different use cases.  The `query-timeout` is intended to automatically kill queries to prevent them from consuming too many resources.  The `show queries` statement is useful for interactive monitoring of queries currently running.

The slow query log setting would is useful to identify slow queries retrospectively.

### PR DESCRIPTION
**Pull request by jsternberg**
Fixes #6429.

### CODE REVIEW

**Comment by jsternberg** [comment]
Example output:

```
[query] 2016/04/20 13:10:03 SELECT count(value) FROM stress."default".cpu
[query] 2016/04/20 13:10:04 Detected slow query: SELECT count(value) FROM cpu (qid: 1, database: stress)
```

Example config:

```
[cluster]
  log-queries-after = "1s"
```

**Comment by e-dard** [comment]
LGTM 👍

File: influxql/query_executor.go [hunk 1]
```diff
@@ -355,6 +359,19 @@ func (e *QueryExecutor) attachQuery(q *Query, database string, interrupt <-chan
...
 
 	go e.waitForQuery(qid, query.closing, interrupt, query.monitorCh)
+	if e.LogQueriesAfter != 0 {
+		go query.monitor(func(closing <-chan struct{}) error {
+			t := time.NewTimer(e.LogQueriesAfter)
+			defer t.Stop()
+
+			select {
+			case <-t.C:
+				e.logger().Printf("Detected slow query: %s (qid: %d, database: %s)", query.query, qid, query.database)
```
Comment (jwilder) [inline influxql/query_executor.go]: Maybe add the threshold to this log message?

Comment (jsternberg) [inline influxql/query_executor.go]: Done.

**Comment by jwilder** [comment]
:+1:

**Comment by adrianlzt** [comment]
Shouldn't be this option in the config file installed by default?

**Comment by somnmos** [comment]
how to print the slow query log to log file?

**Pull request merged**